api/service/networkinfo/service.go

```diff
@@ -220,8 +225,13 @@ func (s *Service) DoService() {
 				utils.Logger().Error().Err(err).Msg("FindPeers")
 				return
 			}
+			if peerInterval < maxFindPeerInterval {
+				peerInterval *= 2
+				intervalTick.Stop()
+				intervalTick = time.NewTicker(time.Duration(peerInterval) * time.Second)
+			}
 
-			s.findPeers(ctx)
+			go s.findPeers(ctx)
```

Comment:
Why to add a goroutine here?

Reply:
don't want findPeers block this function. s.PeerInfo is a separate channel returned by libp2p.

Reply:
I see. So is it true that `findPeers` is never returned and the number of goroutine of findPeers is always increasing? (for channel will never end until the channel is closed)